src/posix_regex.h

```diff
@@ -15,7 +15,7 @@
  * compatible implementation.
  */
 
-#ifdef GIT_REGEX_PCRE
+#ifdef GIT_REGEX_BUILTIN
```

Comment:
Huh? Where do we define `p_regex_` stuff for `GIT_REGEX_PCRE` now?

Edit: never mind, the following commit clears that up